gensim/models/wrappers/fasttext.py

```diff
@@ -298,8 +299,16 @@ def load_dict(self, file_handle, encoding='utf8'):
                 char_byte = file_handle.read(1)
             word = word_bytes.decode(encoding)
             count, _ = self.struct_unpack(file_handle, '@qb')
-            assert self.wv.vocab[word].index == i, 'mismatch between gensim word index and fastText word index'
-            self.wv.vocab[word].count = count
+
+            if word != "__label__":
+                self.wv.vocab[word] = Vocab(index=i, count=count)
```

Comment:
Yes, I understand we have to ready the bytes, agreed.
In that case, I'd prefer something more explicit and clear, like - 
```python
if i == nwords and i < vocab_size:
    assert word == "__label__"
   continue   # don't add word to vocab